src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs

```diff
@@ -537,6 +554,403 @@ public CorElementType PeekElementType()
+        /// <param name="builder">Output string builder</param>
+        /// <param name="owningTypeOverride">Explicit owning type override</param>
+        private TMethod ParseMethodRefToken(TType owningTypeOverride)
+        {
+            uint rid = ReadUInt();
+            return _provider.GetMethodFromMemberRef(_metadataReader, MetadataTokens.MemberReferenceHandle((int)rid), owningTypeOverride);
+        }
+
+    }
+    public class TextSignatureDecoderContext
+    {
+        public TextSignatureDecoderContext(IAssemblyResolver options)
+        {
+            Options = options;
+        }
+
+        public StringBuilder CurrentBuilder;
+        /// <summary>
+        /// Dump options are used to specify details of signature formatting.
+        /// </summary>
+        public IAssemblyResolver Options { get; }
+    }
+
+    /// <summary>
+    /// Helper class used as state machine for decoding a single signature.
+    /// </summary>
+    public class SignatureDecoder : R2RSignatureDecoder<string, string, TextSignatureDecoderContext>
+    {
+        private class TextTypeProvider : StringTypeProviderBase<TextSignatureDecoderContext>, IR2RSignatureTypeProvider<string, string, TextSignatureDecoderContext>
+        {
+            private TextTypeProvider()
+            {
+            }
+
+            public static readonly TextTypeProvider Singleton = new TextTypeProvider();
+
+            public override string GetGenericMethodParameter(TextSignatureDecoderContext genericContext, int index)
+            {
+                return $"mvar #{index}";
+            }
+
+            public override string GetGenericTypeParameter(TextSignatureDecoderContext genericContext, int index)
+            {
+                return $"var #{index}";
+            }
+
+            public override string GetTypeFromSpecification(MetadataReader reader, TextSignatureDecoderContext genericContext, TypeSpecificationHandle handle, byte rawTypeKind)
+            {
+                return MetadataNameFormatter.FormatHandle(reader, handle);
+            }
+
+            public string GetCanonType()
+            {
+                return "__Canon";
+            }
+
+            public string GetMethodFromMethodDef(MetadataReader reader, MethodDefinitionHandle handle, string owningTypeOverride)
+            {
+                StringBuilder signaturePrefixBuilder = new StringBuilder();
+                uint methodDefToken = (uint)MetadataTokens.GetToken(handle);
+                return MetadataNameFormatter.FormatHandle(
+                    reader,
+                    MetadataTokens.Handle((int)methodDefToken),
+                    namespaceQualified: true,
+                    owningTypeOverride: owningTypeOverride);
+            }
+
+            public string GetMethodFromMemberRef(MetadataReader reader, MemberReferenceHandle handle, string owningTypeOverride)
+            {
+                StringBuilder signaturePrefixBuilder = new StringBuilder();
```

Comment:
It looks like this is a local variable that is not being used

Reply:
Deleted